bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Font.java

```diff
@@ -195,7 +213,18 @@ public FontData[] getFontData() {
 	if (isDisposed()) SWT.error(SWT.ERROR_GRAPHIC_DISPOSED);
 	LOGFONT logFont = new LOGFONT ();
 	OS.GetObject(handle, LOGFONT.sizeof, logFont);
-	return new FontData[] {FontData.win32_new(logFont, device.computePoints(logFont, handle))};
+	float heightInPoints = device.computePoints(logFont, handle);
+	int primaryZoom = DPIUtil.mapDPIToZoom (device._getDPIx());
```

Comment:
I would expect this functionality to be encapsulated in the `Device` class and there is already `Device.getDeviceZoom()` that implements this. Can that method be reused (maybe by increasing visibility to be package private)?

Reply:
i can directly use it. after some change I ended up with the same implementation as getDeviceZoom without noticing it.

Reply:
Sounds to me as if there is something broken in general:
- `Display` overwrites that method even though `Device` annotates that method as `@nooverride`
- `Device.getDeviceZoom()` does not seem to be used at all
- `Display` reimplemented the method, but the code here uses the `Device` implementation to retrieve the zoom. So if there is a need to retrieve the value as calulcated in `Device`, there probably should be some (additional) method in `Device` providing that value even if you have an instance of `Display`.